include/inviwo/core/properties/multifileproperty.h

```diff
@@ -77,35 +74,29 @@ class IVW_CORE_API MultiFileProperty : public TemplateProperty<std::vector<std::
                       InvalidationLevel invalidationLevel = InvalidationLevel::InvalidOutput,
                       PropertySemantics semantics = PropertySemantics::Default);
 
-    MultiFileProperty(const MultiFileProperty& rhs);
+    MultiFileProperty(const MultiFileProperty& rhs) = default;
     MultiFileProperty& operator=(const std::vector<std::filesystem::path>& value);
     virtual MultiFileProperty* clone() const override;
     virtual ~MultiFileProperty() = default;
 
     void set(const std::filesystem::path& value);
-    virtual void set(const std::vector<std::filesystem::path>& values) override;
+    void set(const std::vector<std::filesystem::path>& values);
+    void set(const std::vector<std::filesystem::path>& files,
+             const FileExtension& selectedExtension);
+    void set(const MultiFileProperty* property);
+    void set(const FileProperty* property);
     virtual void set(const Property* property) override;
 
-    virtual void serialize(Serializer& s) const override;
-    virtual void deserialize(Deserializer& d) override;
-
-    virtual void addNameFilter(std::string_view filter);
-    virtual void addNameFilter(FileExtension ext);
-    virtual void addNameFilters(const std::vector<FileExtension>& filters);
-    virtual void clearNameFilters();
-    virtual std::vector<FileExtension> getNameFilters();
+    operator const std::vector<std::filesystem::path>&() const { return files_; }
+    const std::vector<std::filesystem::path>& get() const { return files_; }
+    const std::vector<std::filesystem::path>& operator*() const { return files_; };
+    const std::vector<std::filesystem::path>* operator->() const { return &files_.value; }
 
-    virtual void setAcceptMode(AcceptMode mode);
-    AcceptMode getAcceptMode() const;
+    const std::filesystem::path* front() const;
```

Comment:
Why pointers?

Reply:
What about begin(), end()?

Reply:
since it might be empty... you can get the whole list and do that ever... this is for connivence in a few places... could remove back, since I don't use that. but it looked nice together